src/components/CartComponents/AddToCartButton.js

```diff
@@ -0,0 +1,91 @@
+import React, { useEffect, useState } from "react";
+import { useAuth } from "../../context/AuthContext/AuthContext";
+import { useCartList } from "../../context/CartContext/CartContext";
+import makeApiCall from "../../server/server.request";
+import { createToastMessageList } from "../../Utilities/UtilityFunctions";
+
+const AddToCartButton = ({
+  id,
+  toastMessageList,
+  setToastMessageList,
+  classes,
+}) => {
+  let { cartList, cartDispatch } = useCartList();
+  const [added, setAdded] = useState(false);
+  const { user } = useAuth();
+
+  useEffect(() => {
+    const addedInCart = cartList.find((item) => item.id === id);
+
+    if (addedInCart) {
+      setAdded(true);
+    }
+  }, [cartList]);
+
+  const handleAddToCart = async (
+    item,
+    cartList,
+    cartDispatch,
+    toastMessageList,
+    setToastMessageList
+  ) => {
+    let isPresentInCart = cartList.find((itemInCart) => itemInCart.id === item);
+    if (isPresentInCart === undefined || isPresentInCart === null) {
```

Comment:
No need to explicitly check for a falsy value
```suggestion
    if (isPresentInCart) {
```

Reply:
this is fixed